src-java/stats-topology/stats-storm-topology/src/test/java/org/openkilda/wfm/topology/stats/StatsTopologyTest.java

```diff
@@ -426,6 +494,519 @@ public void flowStatsTest() {
+        validateHaFlowStats(reverseEgress, cookieReverse, SWITCH_ID_1, false,
+                true, false, SUB_FLOW_ID_SHARED);
+    }
+
+    /**
+     * 2 switch ha-flow meter-stats test.
+     * 1----2  (y-point 2) where 2 is an y-point switch and 1 is an ingress for forward direction
+     */
+    @Test
+    public void haFlowMeterStatsTest4() {
+        HaFlow haFlow = createHaFlow4();
+        sendUpdateHaFlowPathInfo(haFlow);
+        //forward
+        MeterStatsEntry forward = new MeterStatsEntry(
+                METER_ID_HA_FLOW_PATH_FORWARD.getValue(), 400L, 500L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_1, Collections.singletonList(forward)));
+        validateHaFlowMeterStats(forward, haFlow.getForwardPath().getCookie(), SWITCH_ID_1,
+                METER_ID_HA_FLOW_PATH_FORWARD.getValue(), SUB_FLOW_ID_SHARED, false);
+
+        //reverse
+        MeterStatsEntry reverse1 = new MeterStatsEntry(
+                METER_ID_HA_FLOW_PATH_REVERSE.getValue(), 400L, 500L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_2, Collections.singletonList(reverse1)));
+        validateHaFlowMeterStats(reverse1, haFlow.getReversePath().getCookie(),
+                SWITCH_ID_2, METER_ID_HA_FLOW_PATH_REVERSE.getValue(), null, true);
+    }
+
+    /**
+     * 2 switch ha-flow flow-stats test.
+     * 1----2   where 1 is an y-point switch and 2 is an egress for forward direction
+     */
+    @Test
+    public void haFlowStatsTest5() {
+        HaFlow haFlow = createHaFlow5();
+        sendUpdateHaFlowPathInfo(haFlow);
+        //forward
+        FlowStatsEntry forwardYPoint = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_1, Collections.singletonList(forwardYPoint)));
+        validateHaFlowStats(forwardYPoint, haFlow.getForwardPath().getCookie(), SWITCH_ID_1, true,
+                true, true, SUB_FLOW_ID_SHARED);
+
+        FlowStatsEntry forwardEgress = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(forwardEgress)));
+        validateHaFlowStats(forwardEgress, haFlow.getForwardPath().getCookie().toBuilder()
+                        .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build(), SWITCH_ID_2, false,
+                true, true, SUB_FLOW_ID_SHARED);
+
+
+        //reverse:
+        FlowSegmentCookie cookieReverse = haFlow.getReversePath().getCookie();
+
+        FlowStatsEntry reverseYPoint1 = new FlowStatsEntry(1,
+                cookieReverse.toBuilder().subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(reverseYPoint1)));
```

Comment:
y point is on switch 1

Reply:
is it still relevant for the current test's topology?